File: problems-extend-adapt/0101-0200/0138_copying-the-random-pointer-list/provided/javascript/random_list_node.js
// Problem-provided list node with a random pointer (LC 138 contract).
// The judge's decoder chains next and then sets random by row index.
class RandomListNode {
    constructor(val, next, random) {
        this.val = val ?? 0;
        this.next = next ?? null;
        this.random = random ?? null;
    }
}

File: problems-originals/0101-0200/0138_copy-list-with-random-pointer/provided/rust/random_list_node.rs
// Problem-provided list node with a random pointer (LC 138 contract).
// Paths are fully qualified: every assembled source shares one module.
pub struct RandomListNode {
    pub val: i32,
    pub next: Option<std::rc::Rc<std::cell::RefCell<RandomListNode>>>,
    pub random: Option<std::rc::Rc<std::cell::RefCell<RandomListNode>>>,
}

impl RandomListNode {
    pub fn new(val: i32) -> Self {
        RandomListNode {
            val,
            next: None,
            random: None,
        }
    }
}
